### PR TITLE
Benchmark execution layer

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

Add support to just benchmark the execution layer without storage.
This also allows us to use a mock in memory store where we could track number of DB reads per tx.

## Test Plan 

CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-single-node-benchmark/src/command.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ pub struct Command {
 
 #[derive(Copy, Clone, EnumIter, ValueEnum)]
 pub enum Component {
+    ExecutionOnly,
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/command.rs:41]: Mention this uses in memory store?

File: crates/sui-single-node-benchmark/src/single_node.rs [hunk 2]
```diff
@@ -161,12 +159,51 @@ impl SingleValidator {
...
-            Component::TxnSigning => unreachable!(),
+            Component::TxnSigning | Component::CheckpointExecutor | Component::ExecutionOnly => {
+                unreachable!()
+            }
         };
         assert!(effects.status().is_ok());
         effects
     }
 
+    pub(crate) async fn execute_transaction_in_memory(
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/single_node.rs:170]: IIUC, this function avoids reading object from rocksdb, but still writes output to rocksdb? Impressive that it already improves throughput this much.

**Review by mwtian — APPROVED** [review]

File: crates/sui-single-node-benchmark/src/single_node.rs (hunk 2 above)
Comment (lxfind) [reply crates/sui-single-node-benchmark/src/single_node.rs:170]: No, it doesn't interact with rocksdb at all.

Comment (mwtian) [reply crates/sui-single-node-benchmark/src/single_node.rs:170]: Got it, this skips committing effects and other results.

**Pull request merged**